jwst/assign_wcs/util.py

```diff
@@ -390,159 +423,210 @@ def create_grism_bbox(input_model, reference_files,
 
     Notes
     -----
-    The wavelengthrange reference file is used to govern the extent of the
-    bounding box for each object. The name of the catalog has been stored
-    in the input models meta information under the source_catalog key.
+    The wavelengthrange reference file is used to govern
+    the extent of the bounding box for each object. The name of the
+    catalog has been stored in the input models meta information under
+    the source_catalog key.
 
-    It's left to the calling routine to cut the bounding boxes at the extent
-    of the detector (for example, extract 2d would only extract the on-detector
-    portion of the bounding box)
+    It's left to the calling routine to cut the bounding boxes at the
+    extent of the detector (for example, extract 2d would only extract
+    the on-detector portion of the bounding box)
 
-    Bounding box dispersion direction is dependent on the filter and module for
-    NIRCAM and changes for GRISMR, but is consistent for GRISMC, see
-    https://jwst-docs.stsci.edu/display/JTI/NIRCam+Wide+Field+Slitless+Spectroscopy
+    Bounding box dispersion direction is dependent on the filter and
+    module for NIRCAM and changes for GRISMR, but is consistent for
+    GRISMC, see https://jwst-
+    docs.stsci.edu/display/JTI/NIRCam+Wide+Field+Slitless+Spectroscopy
 
-    NIRISS only has one detector, but GRISMC disperses along rows and GRISMR disperses
-    along columns.
+    NIRISS only has one detector, but GRISMC disperses along rows and
+    GRISMR disperses along columns.
+
+    photutils is used to create the source catalog in the pipeline,
+    it's currently using the fits wcs for it's sky translation, it
+    grabs the fits wcs from the data model using get_fits_wcs(). Until
+    GWCS is fully implemented, this translation may be off in the
+    pipeline.
 
     """
-    # figure out the dispersion direction, shouldn't this be in the polynomials already?
-    disperse_row_right = True  # disperse to increasing x
-    disperse_column = False  # column always disperses to increasing y
+    if use_fits_wcs:
+        log.info("Using fits_wcs object for sky->detector translations")
+
+    if not isinstance(mmag_extract, (int, float)):
+        raise TypeError("Expected mmag_extract to be an int or float")
+    log.info("Extracting objects < abmag = {0}".format(mmag_extract))
+
+    if extract_orders is not None:
+        if isinstance(extract_orders, list):
+            if not all(isinstance(item, int) for item in extract_orders):
+                raise TypeError("Expected extract_orders to be a list of ints")
+        else:
+            raise TypeError("Expected extract_orders to be a list of ints")
```

Comment:
It looks like the check for the type of `extract_orders` is repeated below starting on line 512. Remove this one?

Reply:
ah, yah, that was also from the bad merge, thanks for catching :)